contracts/dtoken/src/supply.rs

```diff
@@ -4,23 +4,30 @@ use crate::*;
 impl Contract {
     #[payable]
     pub fn supply(&mut self, token_amount: WBalance) -> PromiseOrValue<U128> {
+        let user_profile = UserProfileDtoken::default();
+        let supply_amount = user_profile.get_profile(self.get_signer_address().clone()).supply_amount as u128;
+        if user_profile.is_exist(self.get_signer_address().clone()) &&
```

Comment:
@TymofiiSavinov 
we should create profile on Supply (if doesn't exist) 
and we should prevent Withdraw, Borrow, Repay if user profile doens't exist